spring-integration-core/src/main/java/org/springframework/integration/store/MessageMetadata.java

```diff
@@ -29,10 +29,14 @@
 
 	private static final long serialVersionUID = 1L;
 
-	private final UUID messageId;
+	private UUID messageId;
```

Comment:
It should be `volatile` if it potentially can be modified concurrently.

Frankly we have stopped to add `volatile` to those properties which mostly are modified only during initialization for a while already.

There is really some performance impact from the `volatile`. Can't find the link to discussion, but I recall that really was from you do not abuse `volatile`...